dask/threaded.py

```diff
@@ -536,7 +543,7 @@ def fire_task():
                 fire_task()
 
         # Main loop, wait on tasks to finish, insert new ones
-        while state['waiting'] or state['ready'] or len(state['finished']) < len(jobs):
+        while state['waiting'] or state['ready'] or (len(state['finished']) < len(jobs)):
```

Comment:
I realize that.
The parens were added to make the association explicit (to the reader of
the code).
Upon my first reading of the condition, I wasn't 100% certain of the
intent, and the extra parens make it clearer to me.



On Tue, Jan 13, 2015 at 12:43 PM, Phillip Cloud <notifications@github.com>
wrote:

> In dask/threaded.py
> <https://github.com/ContinuumIO/dask/pull/6#discussion-diff-22879205>:
>
> > @@ -536,7 +543,7 @@ def fire_task():
> >                  fire_task()
> >
> >          # Main loop, wait on tasks to finish, insert new ones
> > -        while state['waiting'] or state['ready'] or len(state['finished']) < len(jobs):
> > +        while state['waiting'] or state['ready'] or (len(state['finished']) < len(jobs)):
>
> these parens aren't necessary, < has higher precedence than or
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ContinuumIO/dask/pull/6/files#r22879205>.
>